src/main.rs

```diff
@@ -1,3 +1,208 @@
-fn main() {
+use chrono::NaiveDateTime;
+use clap::Parser;
+use serde::{Deserialize, Serialize};
 
+use std::borrow::Cow;
+use std::collections::HashMap;
+use std::path::PathBuf;
+use std::{fs, io};
+
+fn main() -> io::Result<()> {
+    let cli = Cli::parse();
+
+    let content = fs::read_to_string(cli.file)?;
+
+    let chat: Chat = serde_json::from_str(&content).unwrap();
+    let stat: ChatStatistics = ChatStatistics::gather(&chat, cli.jobs);
+
+    if let Some(output) = cli.output {
+        let file = fs::File::create(output)?;
+        serde_json::to_writer_pretty(file, &stat)?;
+    }
+
+    Ok(())
+}
+
+#[derive(Debug, Parser)]
+struct Cli {
+    #[arg(long, short)]
+    file: PathBuf,
+    #[arg(long, short, default_value = "out.json")]
+    output: Option<PathBuf>,
```

Comment:
if you have a default value, then you don't need `Option` here

```suggestion
    output: PathBuf,
```